AppleALC/ALCUserClientProvider/ALCUserClientProvider.cpp

```diff
@@ -0,0 +1,81 @@
+//
+//  ALCUserClientProvider.cpp
+//  AppleALC
+//
+//  Created by Nick on 10/14/20.
+//  Copyright © 2020 vit9696. All rights reserved.
+//
+
+#include "ALCUserClientProvider.hpp"
+
+OSDefineMetaClassAndStructors(ALCUserClientProvider, IOService);
+
+bool ALCUserClientProvider::start(IOService* provider) {
+	if (!super::start(provider))
+		return false;
+	
+	mHDACodecDevice = IOService::waitForMatchingService(IOService::nameMatching(kIOHDACodecDevice), 3000);	// Wait for 3s
+	
+	if (!mHDACodecDevice)
+	{
+		DBGLOG("client", "HDACodecDevice not found");
+		stop(provider);
+		return false;
+	}
+	
+	// Only attach to internal codec
+	auto codecAddress = OSDynamicCast(OSNumber, mHDACodecDevice->getProperty(kIOHDACodecAddress));
+	if (!codecAddress) {
+		DBGLOG("client", "IOHDACodec address is non-existent");
+		stop(provider);
+		return false;
+	}
+	
+	// Check if attached to correct IOHDACodecDevice
+	if (codecAddress->unsigned8BitValue() != 0) {
```

Comment:
I am pretty sure I saw HDEFs with non-zero address, so this check is not quite right.

Reply:
Had my doubts about it. Checked on 2 systems and found the address to be `0`. Anyways, removed the check.